cpp/daal/src/externals/service_math_ref.h

```diff
@@ -58,48 +58,88 @@ struct RefMath<double, cpu>
 
     static double sPowx(double in, double in1) { return pow(in, in1); }
 
+    static double xsPowx(double in, double in1) { return pow(in, in1); }
+
     static double sCeil(double in) { return ceil(in); }
 
+    static double xsCeil(double in) { return ceil(in); }
+
     // Not implemented
     static double sErfInv(double in) { return std::numeric_limits<double>::quiet_NaN(); }
 
+    // Not implemented
+    static double xsErfInv(double in) { return std::numeric_limits<double>::quiet_NaN(); }
+
     static double sErf(double in) { return erf(in); }
 
+    static double xsErf(double in) { return erf(in); }
+
     static double sLog(double in) { return log(in); }
 
+    static double xsLog(double in) { return log(in); }
+
     // Not implemented
     static double sCdfNormInv(double in) { return std::numeric_limits<double>::quiet_NaN(); }
 
+    // Not implemented
+    static double xsCdfNormInv(double in) { return std::numeric_limits<double>::quiet_NaN(); }
+
     static void vPowx(SizeType n, const double * in, double in1, double * out)
     {
 #pragma omp simd
         for (SizeType i = 0; i < n; ++i) out[i] = pow(in[i], in1);
     }
 
+    static void xvPowx(SizeType n, const double * in, double in1, double * out)
+    {
+        for (SizeType i = 0; i < n; ++i) out[i] = pow(in[i], in1);
+    }
+
     static void vCeil(SizeType n, const double * in, double * out)
     {
 #pragma omp simd
         for (SizeType i = 0; i < n; ++i) out[i] = ceil(in[i]);
     }
 
+    static void xvCeil(SizeType n, const double * in, double * out)
+    {
+        for (SizeType i = 0; i < n; ++i) out[i] = ceil(in[i]);
+    }
+
     // Not implemented
     static void vErfInv(SizeType n, const double * in, double * out)
     {
         for (SizeType i = 0; i < n; ++i) out[i] = std::numeric_limits<double>::quiet_NaN();
     }
 
+    // Not implemented
+    static void xvErfInv(SizeType n, const double * in, double * out)
+    {
+        for (SizeType i = 0; i < n; ++i) out[i] = std::numeric_limits<double>::quiet_NaN();
+    }
+
     static void vErf(SizeType n, const double * in, double * out)
     {
 #pragma omp simd
         for (SizeType i = 0; i < n; ++i) out[i] = erf(in[i]);
     }
 
+    static void xvErf(SizeType n, const double * in, double * out)
+    {
+        for (SizeType i = 0; i < n; ++i) out[i] = erf(in[i]);
+    }
+
     static void vExp(SizeType n, const double * in, double * out)
     {
 #pragma omp simd
         for (SizeType i = 0; i < n; ++i) out[i] = exp(in[i]);
     }
 
+    static void xvExp(SizeType n, const double * in, double * out)
+    {
+        for (SizeType i = 0; i < n; ++i) out[i] = exp(in[i]);
```

Comment:
Functions like this would run faster when using multiple threads. MKL runs then multi-threaded as far as I can tell.

Reply:
yes, you are right, but I wanted to add the section about these functions usage in threader_for(multithread for), because in such case to avoid calling multithread function inside the multithread loop we must use xFunc(single thread)